File: JavaScript Code/node.js
/*
* Implementation of a node in Javascript. Carries
* a value element, and points to another node.
*
* @constructor
* @param {Object} value: the value element to be stored.
*/
function Node(value) {
	this.value = value;
	this.next = null;
}

/*
* Will return the value of the node.
*
* @return the value the node carries.
*/
Node.prototype.getValue(){
	return this.value;
}

/*
* Will return if the node points to another node.
*
* @return true if it points to another node and false if not.
*/
Node.prototype.hasNext(){
	return (!this.next==null);
}

/*
* Will return the node being pointed to if its not empty.
*
* @return the node that this node points to if not empty,
* otherwise null.
*/
Node.prototype.getNext(){
	var n = null;
	if (this.hasNext()) {
		n = this.next;
	}
	return n;
}

/*
* Will change the value of the node and return the
* old value.
*
* @param {Object} newValue: the new value to be carried by the node.
* @return the old value of the node.
*/
Node.prototype.setValue(newValue){
	var oldValue = this.value;
	this.value = newValue;
	return oldValue;
}

/*
* Will update the node being pointed to and return the old one.
*
* @param {Node} node: the new node to point to.
* @return the old node.
*/
Node.prototype.setNext(newNext){
	var oldNext = this.next;
	this.next = newNext;
	return oldNext;
}

File: Java Code/Node.java
/**
 * Implementation of a double node in Java. Carries
 * a value, and points to two other nodes.
 */

public class Node {

    private Object value;
    private Node next;

    /**
     * Constructor for Node. Takes in a value to be stored,
     * and sets previous and next pointers to null.
     *
     * @param val: the value to be assigned.
     */
    public Node(Object val) {
        this.value = val;
        next = null;
    }

    /**
     * Returns the value that is stored by the node.
     *
     * @return the value of the node.
     */
    public Object getValue() {
        return this.value;
    }

    /**
     * Returns if the Node points to another node.
     *
     * @return: true if next points to a node, and false if
     * next is null.
     */
    public boolean hasNext() {
        return (next != null);
    }

    /**
     * Returns the pointer of the next Node.
     *
     * @return the node being pointed to by self in front.
     */
    public Node getNext() {
        return this.next;
    }

    /**
     * Sets the node's value, and returns the previous value.
     *
     * @param newValue: the new value to be carried by the node.
     * @return the old value of the node, and null if there is none.
     */
    public Object setValue(Object newValue) {
        Object oldValue = this.value;
        this.value = newValue;
        return oldValue;
    }

    /**
     * Sets the next node being pointed to, and returns the old one.
     *
     * @param newNext: the new node to be pointed to
     * @return the old next, and null if there was no value.
     */
    public Node setNext(Node newNext) {
        Node oldNext = this.next;
        this.next = newNext;
        return oldNext;
    }

}
